src/Game/UI/Gumps/ContainerGump.cs

```diff
@@ -63,10 +63,22 @@ public ContainerGump(Item item, Graphic gumpid) : this()
             foreach (var c in Children.OfType<ItemGump>())
                 c.Dispose();
 
-            foreach (Item i in _item.Items.Where(s => s.ItemData.Layer != (int) Layer.Hair && s.ItemData.Layer != (int) Layer.Beard && s.ItemData.Layer != (int) Layer.Face))
+            foreach (Item i in _item.Items)
+            {
+                if (i == null || 
+                    i.ItemData.Layer == (int) Layer.Hair || 
+                    i.ItemData.Layer == (int) Layer.Beard || 
+                    i.ItemData.Layer == (int) Layer.Face ||
+                    // Skip non-lootable items on Outlands (server always sends their position as 0,0,0)
+                    // FIXME: Remove this check when the issue with non-lootable items will be resolved on the server side
+                    (Engine.GlobalSettings.ShardType == 2 && i.X == 0 && i.Y == 0 && i.Z == 0)
+                )
+                    continue;
```

Comment:
📝 Follow-up to my PR comment: this doesn't seem like it should affect UO:R's monster packs, so I'm on board. However, I'd suggest leaving `ContainerGump` unmodified, start by just making the changes to `GridLootGump`.
Better yet, incorporate a configurable option to enable/disable rendering non-lootable items in the grid loot gump, in case some players would be confused to suddenly stop seeing items appear in the gump which they know should be on the corpse, even if they can't loot those items.

Reply:
You are absolutely right: the `ShardType` check is there to only apply that for Outlands shard where the behavior of non-lootable items is confirmed.

If you check the code closely, then you can see that we check for the exact item position in the container. This position is a dead zone for any container and no items can be placed there manually by a player.